src/frontend/src/binder/insert.rs

```diff
@@ -187,54 +190,82 @@ impl Binder {
                             .collect(),
                     )?,
                 };
+                bound_query.schema().len()
             }
             Some(values) => {
-                assert!(!values.0.is_empty());
-                let num_value_cols = values.0[0].len();
-                let has_user_specified_columns = !cols_to_insert_by_user.is_empty();
-                let num_target_cols = if has_user_specified_columns {
-                    cols_to_insert_by_user.len()
-                } else {
-                    cols_to_insert_in_table.len()
-                };
-                let err_msg = match num_target_cols.cmp(&num_value_cols) {
-                    std::cmp::Ordering::Equal => None,
-                    std::cmp::Ordering::Greater => {
-                        if has_user_specified_columns {
-                            // e.g. insert into t (v1, v2) values (7)
-                            Some("INSERT has more target columns than expressions")
-                        } else {
-                            // e.g. create table t (a int, b real)
-                            //      insert into t values (7)
-                            // this kind of usage is fine, null values will be provided
-                            // implicitly.
-                            None
-                        }
-                    }
-                    std::cmp::Ordering::Less => {
-                        // e.g. create table t (a int, b real)
-                        //      insert into t (v1) values (7, 13)
-                        // or   insert into t values (7, 13, 17)
-                        Some("INSERT has more expressions than target columns")
-                    }
-                };
-                if let Some(msg) = err_msg {
-                    return Err(RwError::from(ErrorCode::BindError(msg.to_string())));
-                }
-
+                let values_len = values
+                    .0
+                    .first()
+                    .expect("values list should not be empty")
+                    .len();
                 let values = self.bind_values(values.clone(), Some(expected_types))?;
```

Comment:
since we decided to punt the NULL padding to the batch executor, perhaps the NULL padding logic in `bing_values` should be deleted

Reply:
True, actually I just forgot to remove deprecated comments :)